src/github.ts

```diff
@@ -452,17 +453,17 @@ export class GitHub {
     preRelease = false
   ): Promise<GitHubTag | undefined> {
     let regexpString = prefix || '';
-    if (!preRelease) regexpString = `${regexpString}[^-]*`;
+    if (!preRelease) regexpString = `^${regexpString}[^-]*$`;
```

Comment:
curious if this addresses a bug you bumped into, or if you're just being prudent by terminating the regex.

Reply:
I'm pretty sure this is in response to a unit test that broke with the non-terminated one.